File: TfgSn.App/src/app/models/Post/PostDto.ts
import { CommentDTO } from "../Comment/CommentDTO";
import { Comments } from "../Interfaces/Comment";
import { LikesDTO } from "../Likes/LikesDto"


export interface PostDto {
    id: number;
    userName: string;
    text: string;
    image: Uint8Array;
    creationDate: Date;
    description: string;
    comments: Comments[];
    likes: LikesDTO[];
    userHasLiked?: boolean;
  }

File: ApiTFG/Persistence/Dtos/PostDTO.cs
using ApiTFG.Migrations;
using System.Runtime.InteropServices;

namespace ApiTFG.Dtos
{
    public class PostDTO
    {
        public int Id { get; set; }
        public string Text { get; set; }
        public byte[] Image { get; set; }
        public DateTime CreationDate { get; set; }
        public ICollection<CommentDTO> Comments { get; set; }
        public ICollection<LikesDTO> Likes { get; set; }

        public PostDTO()
        {
            Comments = new HashSet<CommentDTO>();
            Likes = new HashSet<LikesDTO>();
        }
    }
}
